packages/bpk-component-pagination/src/BpkPaginationPage.js

```diff
@@ -25,32 +25,41 @@ import STYLES from './bpk-pagination-page.scss';
 
 const getClassName = cssModules(STYLES);
 
+const defaultLabelFormatter = (page, isSelected) =>
+  `Go to page ${page}${isSelected ? ', this is the current page' : ''}.`;
+
 const BpkPaginationPage = (props) => {
   const {
-    label, onSelect, isSelected,
+    page, onSelect, isSelected, pageLabelFormatter,
   } = props;
 
+  const pageFormatter = pageLabelFormatter || defaultLabelFormatter;
+
   return (
     <BpkButton
       iconOnly
       secondary
       selected={isSelected}
       onClick={onSelect}
       className={getClassName('bpk-pagination-page')}
+      aria-label={pageFormatter(page, isSelected)}
+      aria-current={isSelected}
     >
-      <span>{ label }</span>
+      <span>{ page }</span>
     </BpkButton>
   );
 };
 
 BpkPaginationPage.propTypes = {
-  label: PropTypes.node.isRequired,
+  page: PropTypes.number.isRequired,
   onSelect: PropTypes.func.isRequired,
   isSelected: PropTypes.bool,
+  pageLabelFormatter: PropTypes.func,
 };
 
 BpkPaginationPage.defaultProps = {
   isSelected: false,
+  pageLabelFormatter: defaultLabelFormatter,
```

Comment:
We don't typically provided a default for these as it makes it possible to miss the translation piece